From d94c40b1e0515117fbe7d04480640641f2a2edbd Mon Sep 17 00:00:00 2001
From: Thomas Bechtold <tbechtold@suse.com>
Date: Tue, 4 Apr 2017 18:09:34 +0200
Subject: [PATCH] Remove deprecated oslo.messaging aliases parameter

Those are remnants from the oslo-incubator times. Also, oslo.messaging
deprecated [1] transport aliases since 5.2.0+ that is the minimal
version supported for stable/newton.

[1] I314cefa5fb1803fa7e21e3e34300e5ced31bba89

Closes-Bug: #1424728
Change-Id: I50c4559ea2ebc8512a05ffad52e5f04b22743ff4
---
 keystonemiddleware/audit/_notifier.py | 20 +-------------------
 1 file changed, 1 insertion(+), 19 deletions(-)

diff --git a/keystonemiddleware/audit/_notifier.py b/keystonemiddleware/audit/_notifier.py
index 83898d4..9d17c29 100644
--- a/keystonemiddleware/audit/_notifier.py
+++ b/keystonemiddleware/audit/_notifier.py
@@ -18,8 +18,6 @@ try:
 except ImportError:
     oslo_messaging = None
 
-from pycadf import cadftaxonomy as taxonomy
-
 from keystonemiddleware.i18n import _LI
 
 
@@ -46,24 +44,8 @@ class _MessagingNotifier(object):
 
 def create_notifier(conf, log):
     if oslo_messaging:
-        proj = conf.project or taxonomy.UNKNOWN
-
-        if proj:
-            # Aliases to support backward compatibility
-            aliases = {
-                '%s.openstack.common.rpc.impl_kombu' % proj: 'rabbit',
-                '%s.openstack.common.rpc.impl_qpid' % proj: 'qpid',
-                '%s.openstack.common.rpc.impl_zmq' % proj: 'zmq',
-                '%s.rpc.impl_kombu' % proj: 'rabbit',
-                '%s.rpc.impl_qpid' % proj: 'qpid',
-                '%s.rpc.impl_zmq' % proj: 'zmq',
-            }
-        else:
-            aliases = {}
-
         transport = oslo_messaging.get_transport(conf.oslo_conf_obj,
-                                                 url=conf.get('transport_url'),
-                                                 aliases=aliases)
+                                                 url=conf.get('transport_url'))
 
         notifier = oslo_messaging.Notifier(
             transport,
-- 
2.12.2

